Xamarin.Forms.Platform.Android/Renderers/SwitchRenderer.cs

```diff
@@ -77,6 +84,42 @@ protected override void OnElementChanged(ElementChangedEventArgs<Switch> e)
 
 				e.NewElement.Toggled += HandleToggled;
 				Control.Checked = e.NewElement.IsToggled;
+				defaultOnColor = Control.TrackTintList;
+				UpdateOnColor();
+			}
+		}
+
+		protected override void OnElementPropertyChanged(object sender, PropertyChangedEventArgs e)
+		{
+			base.OnElementPropertyChanged(sender, e);
+
+			if (e.PropertyName == Switch.OnColorProperty.PropertyName)
+				UpdateOnColor();
+		}
+
+		private void UpdateOnColor()
+		{
+			System.Diagnostics.Debug.WriteLine("Done");
+			if (Element != null)
+			{
+				if (Element.OnColor == Color.Default)
+				{
+					Control.TrackTintList = defaultOnColor;
+				}
+				else
+				{
+					if (Build.VERSION.SdkInt >= BuildVersionCodes.JellyBean)
+					{
+						if (Control.Checked)
+						{
+							Control.TrackDrawable.SetColorFilter(Element.OnColor.ToAndroid(), PorterDuff.Mode.Multiply);
```

Comment:
This seems odd since everything else sets the TrackTintList. Further PorterDuff multiply isn't going to work at all for dark themes as 0 times anything is still zero :/

In would probably be a better mode. However not using the ColorFilter at all would be best.